fixes.inc

```diff
@@ -14281,6 +14411,242 @@ native BAD_DestroyActor(actorid) = DestroyActor;
+				}
+			#endif
+		}
+		return ret;
+	}
+
+	stock FIXES_RemoveVehicleComponent(vehicleid, componentid)
+	{
+		new const ret = RemoveVehicleComponent__(vehicleid, componentid);
+		if(ret == 0) 
+		{
+			return ret;
+		}
+		switch(FIXES_GetVehicleComponentType(componentid))
+		{
+			#if !FIXES_Single
+				case CARMODTYPE_STEREO:
+				{
+					setproperty(.name = FIXES_gscStereo, .value = 0);
+				}
+				case CARMODTYPE_FRONT_BUMPER:
+				{
+					setproperty(.name = FIXES_gscFrontBumper, .value = 0);
+				}
+				case CARMODTYPE_REAR_BUMPER:
+				{
+					setproperty(.name = FIXES_gscRearBumper, .value = 0);
+				}
+				case CARMODTYPE_FRONT_BULLBAR:
+				{
+					setproperty(.name = FIXES_gscFrontBullBar, .value = 0);
+				}
+				case CARMODTYPE_REAR_BULLBAR:
+				{
+					setproperty(.name = FIXES_gscRearBullbar, .value = 0);
+				}
+			#else 
+				case CARMODTYPE_STEREO:
+				{
+					FIXES_gsStereo[vehicleid] = 0;
+				}
+				case CARMODTYPE_FRONT_BUMPER:
+				{
+					FIXES_gsFrontBumper[vehicleid] = 0;
+				}
+				case CARMODTYPE_REAR_BUMPER:
+				{
+					FIXES_gsRearBumper[vehicleid] = 0;
+				}
+				case CARMODTYPE_FRONT_BULLBAR:
+				{
+					FIXES_gsFrontBullBar[vehicleid] = 0;
+				}
+				case CARMODTYPE_REAR_BULLBAR:
+				{
+					FIXES_gsRearBullbar[vehicleid] = 0;
+				}
+			
+			#endif
+		}
+		return ret;
+	}
+
+	stock FIXES_GetVehicleComponentInSlot(vehicleid, slot)
+	{
+		switch(slot)
+		{
+			#if !FIXES_Single
+				case CARMODTYPE_STEREO:
+				{
+					return getproperty(.name = FIXES_gscStereo);
+				}
+				case CARMODTYPE_FRONT_BUMPER:
+				{
+					return getproperty(.name = FIXES_gscFrontBumper);
+				}
+				case CARMODTYPE_REAR_BUMPER:
+				{
+					return getproperty(.name = FIXES_gscRearBumper);
+				}
+				case CARMODTYPE_FRONT_BULLBAR:
+				{
+					return getproperty(.name = FIXES_gscFrontBullBar);
+				}
+				case CARMODTYPE_REAR_BULLBAR:
+				{
+					return getproperty(.name = FIXES_gscRearBullbar);
+				}
+			#else 
+				case CARMODTYPE_STEREO:
+				{
+					return FIXES_gsStereo[vehicleid];
+				}
+				case CARMODTYPE_FRONT_BUMPER:
+				{
+					return FIXES_gsFrontBumper[vehicleid];
+				}
+				case CARMODTYPE_REAR_BUMPER:
+				{
+					return FIXES_gsRearBumper[vehicleid];
+				}
+				case CARMODTYPE_FRONT_BULLBAR:
+				{
+					return FIXES_gsFrontBullBar[vehicleid];
+				}
+				case CARMODTYPE_REAR_BULLBAR:
+				{
+					return FIXES_gsRearBullbar[vehicleid];
+				}
+			#endif 
+		}
+		return GetVehicleComponentInSlot__(vehicleid, slot);
+	}
+	#if _FIXES_SAMP
+		#define _ALS_GetVehicleComponentInSlot
+		#define GetVehicleComponentInSlot( FIXES_GetVehicleComponentInSlot(
+		#define _ALS_AddVehicleComponent
+		#define AddVehicleComponent( FIXES_AddVehicleComponent(
+		#define _ALS_RemoveVehicleComponent
+		#define RemoveVehicleComponent( FIXES_RemoveVehicleComponent(
+		#define _ALS_DestroyVehicle
+		#define DestroyVehicle( FIXES_DestroyVehicle(
+	#endif
+
+	#define _ALS_GetVehicleComponentInSlot__
```

Comment:
All these functions should be included separately, in their own conditional compilation block with their own header comments.